src/include/transaction/transaction_manager.h

```diff
@@ -92,6 +95,14 @@ class TransactionManager {
    */
   std::queue<std::pair<timestamp_t, Action>> DeferredActionsForGC();
 
+  /**
+   * Installs a constraint so that future committing transactions will verify with it
+   * @param txn context of transaction installing this constraint
+   * @param fn the function ran to verify the constraint
+   * @return pointer to the installed TransactionConstraint
+   */
+  TransactionConstraint *InstallConstraint(TransactionContext *txn, constraint_fn fn);
```

Comment:
Why does this need to return the constraint object?  It seems like the plan is that the constraint objects live in and are managed by the TransactionManager which is why you have the `InstallingTransactionID` function in its API.

Reply:
This is mainly for tests, otherwise I would have to expose the entire constraint list for the tests to access which seems worse.

Reply:
I ended up removing it since the outside user shouldn't see anything related to enforcing or not, so tests shouldn't be looking at it either